app/popup/popup.js

```diff
@@ -196,6 +194,112 @@ function reloadPopup(tabID) {
       request.showLazyloadedImages();
     });
 
+    document.getElementById("parent-list").addEventListener("click", function (e) {
+      if (e.target && e.target.nodeName == "INPUT") {
+        jeclique(e.target.id);
+      }
+    });
+    
+    function jeclique(cookieName) {
+      var toggle = true;
+      toggle = document.getElementById(cookieName).checked;
+      if (toggle == false) {
+        browser.tabs.query({ active: true, lastFocusedWindow: true }).then(tabs => {
+          let url = tabs[0].url.split('?')
+          let base = url[0];
+          let params = url.length > 1 ? url[1] : null
+
+          m_dict = this.getParamsDict(cookieName, false)
+          p_dict = {}
+          if (params != null) {
+            p_dict = this.getParamsDict(params);
+          }
+
+          params = { ...p_dict, ...m_dict };
+          params = this.convertObjectToUrlString(params)
+          newUrl = base + params
+          browser.tabs.update(tabID, { url: newUrl })
+        });
+      }
+      else if (toggle == true) {
+        browser.tabs.query({ active: true, lastFocusedWindow: true }).then(tabs => {
+          let url = tabs[0].url.split('?')
+          let base = url[0];
+          let params = url.length > 1 ? url[1] : null
+
+          m_dict = this.getParamsDict(cookieName, true)
+          p_dict = {}
+          if (params != null) {
+            p_dict = this.getParamsDict(params);
+          }
+          params = { ...p_dict, ...m_dict };
+          params = this.convertObjectToUrlString(params)
+          newUrl = base + params
+          browser.tabs.update(tabID, { url: newUrl })
+        });
+      }
+    }
+
+    convertObjectToUrlString = (obj) => {
+      let url = "?"
+      for (let [key, value] of Object.entries(obj)) {
+        url += `${key}=${value}&`;
+      }
+      if (url.length > 1) {
+        url = url.substr(0, url.length - 1);
+      }
+      return url;
+    }
+
+    getParamsDict = (p_string, p_value) => {
+      if (p_value == false || p_value == true) {
```

Comment:
this method is duplicated. The only change is `p_dict[param[0]] = p_value` and `p_dict[param[0]] = param[1]`